owiddata/generate-owiddata-stats.py

```diff
@@ -48,22 +77,91 @@ def main(args):
                                                                        "total_vaccinations"].item()/1000000))) + \
                                        " million"
-    owid_stats["daily_vax_rate"] = str("{:,}".format(round(vaccine_nums[vaccine_nums["location"] ==
+    owid_stats["owid_daily_rate"] = str("{:,}".format(round(vaccine_nums[vaccine_nums["location"] ==
                                                 "World"].loc[vaccine_nums["date"] ==
-                                                             owid_stats["most_recent_date"],
+                                                             owid_stats["owid_most_recent_date"],
                                                              "daily_vaccinations_per_million"].item()))) + " per million"
-    owid_stats["total_countries"] = format(vaccine_locations["location"].nunique())
+    owid_stats["owid_total_countries"] = format(vaccine_locations["location"].nunique())
 
     # Identify number of vaccine manufacturers included in location totals (not the same as manufacturer-specific data)
-    owid_stats["vaccine_types"] = format(len(set([item.strip() for countryList in vaccine_locations["vaccines"].to_list() for
-                                           item in countryList.split(",")])))
+    vaxTypes = set([item.strip() for countryList in vaccine_locations["vaccines"].to_list() for item in countryList.split(",")])
+    owid_stats["owid_vaccine_types"] = format(len(vaxTypes))
+    vaxPlatforms = pd.read_csv(args.platform_types, index_col="Manufacturer")
+
+    # Count the number of vaccines being administered per technology type
+    numVax = vaxPlatforms["Type"].value_counts()
+
+    # Set the parameters color-coding the plots. Scale is the max candidates adminstered across all vaccine types.
+    scale = max(numVax)
+    cmap = mpl.cm.Purples
+    norm = mpl.colors.BoundaryNorm(np.arange(0, scale + 1), cmap.N)
+
+    # Check that platform information is present (needs to be manually determined and input in vaccine_platforms.csv
+    missingInfo = [vax for vax in vaxTypes if vax not in vaxPlatforms.index]
+    if len(missingInfo) > 0:
+        exit("Missing platform information for " + ", ".join(missingInfo))
+
+    # Transform list of vaccine candidate per iso code to list of ISO codes per vaccine candidate
+    allVaxByCountry = dict(zip(vaccine_locations["iso_code"],
+                               vaccine_locations["vaccines"]))
+    countryByVax = dict()
+    for iso, vaccines in allVaxByCountry.items():
+        for vax in vaccines.split(","):
+            vax = vax.strip()
+            countryCodes = countryByVax.get(vax, [])
+            countryByVax[vax] = countryCodes + [iso]
+
+    # Add countries to vaccine platform info and plot each vaccine type
+    vaxPlatforms['countries'] = vaxPlatforms.index.map(countryByVax)
+
+    for platform in set(list(vaxPlatforms["Type"])):
```

Comment:
```suggestion
    for platform in set(vaxPlatforms["Type"]):
```
Not a big deal, but is the `list()` needed here?

Reply:
Nope, it's not needed! For some reason I did that while debugging. I'll push a new commit to remove it.